server/rest/dataset.py

```diff
@@ -530,3 +532,35 @@ def createTweetGeocodes(self, item, params):
         .param('id', 'The Item ID', paramType='path')
         .errorResponse('ID was invalid.')
         .errorResponse('Write permission denied on the Item.', 403))
+
+    @access.public
+    @loadmodel(map={'wmsSourceId': 'wmsSource'}, model='item',
+               level=AccessType.READ)
+    def createWmsDataset(self, wmsSource, params):
```

Comment:
Currently we have one file dataset.py that creates various dataset. As we add more dataset types this will get bigger and bigger. My suggestion with creating multiple files / classes will make this file smaller.